# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/wyciwyg/nsWyciwygChannel.cpp
# Commit: 74f99033251c
# Full Hash: 74f99033251cd6c36eb6b0b5a4713eb6b5c793fa
# Author: Jonathan Kingston <jkt@mozilla.com>
# Date: 2019-02-13 09:48:26
# Regressor Bug: 1520868
# File Overlap Count: 1
# Description:
#   Bug 1520868 - Replacing AsyncOpen2 with AsyncOpen always r=valentin
#   
#   Replacing js and text occurences of asyncOpen2
#   Replacing open2 with open
#   
# ==============================================================================

diff -r 3378fd5f4f22 -r 74f99033251c netwerk/protocol/wyciwyg/nsWyciwygChannel.cpp
--- a/netwerk/protocol/wyciwyg/nsWyciwygChannel.cpp	Tue Feb 12 17:58:19 2019 +0100
+++ b/netwerk/protocol/wyciwyg/nsWyciwygChannel.cpp	Tue Feb 12 16:08:25 2019 +0000
@@ -291,28 +291,25 @@
 }
 
 NS_IMETHODIMP
-nsWyciwygChannel::Open(nsIInputStream **aReturn) {
-  return NS_ERROR_NOT_IMPLEMENTED;
-}
-
-NS_IMETHODIMP
-nsWyciwygChannel::Open2(nsIInputStream **aStream) {
+nsWyciwygChannel::Open(nsIInputStream **aStream) {
   nsCOMPtr<nsIStreamListener> listener;
   nsresult rv =
       nsContentSecurityManager::doContentSecurityCheck(this, listener);
   NS_ENSURE_SUCCESS(rv, rv);
-  return Open(aStream);
+
+  return NS_ERROR_NOT_IMPLEMENTED;
 }
 
 NS_IMETHODIMP
-nsWyciwygChannel::AsyncOpen(nsIStreamListener *listener, nsISupports *ctx) {
-  MOZ_ASSERT(
-      !mLoadInfo || mLoadInfo->GetSecurityMode() == 0 ||
-          mLoadInfo->GetInitialSecurityCheckDone() ||
-          (mLoadInfo->GetSecurityMode() ==
-               nsILoadInfo::SEC_ALLOW_CROSS_ORIGIN_DATA_IS_NULL &&
-           nsContentUtils::IsSystemPrincipal(mLoadInfo->LoadingPrincipal())),
-      "security flags in loadInfo but asyncOpen2() not called");
+nsWyciwygChannel::AsyncOpen(nsIStreamListener *aListener) {
+  nsCOMPtr<nsIStreamListener> listener = aListener;
+  nsresult rv =
+      nsContentSecurityManager::doContentSecurityCheck(this, listener);
+  if (NS_FAILED(rv)) {
+    mIsPending = false;
+    mCallbacks = nullptr;
+    return rv;
+  }
 
   LOG(("nsWyciwygChannel::AsyncOpen [this=%p]\n", this));
   MOZ_ASSERT(mMode == NONE, "nsWyciwygChannel already open");
@@ -327,7 +324,7 @@
   // mIsPending set to true since OnCacheEntryAvailable may be called
   // synchronously and fails when mIsPending found false.
   mIsPending = true;
-  nsresult rv = OpenCacheEntryForReading(mURI);
+  rv = OpenCacheEntryForReading(mURI);
   if (NS_FAILED(rv)) {
     LOG(("nsWyciwygChannel::OpenCacheEntryForReading failed [rv=%" PRIx32 "]\n",
          static_cast<uint32_t>(rv)));
@@ -340,26 +337,12 @@
   // we get to this line in case OnCacheEntryAvailable is invoked
   // synchronously.
   mListener = listener;
-  mListenerContext = ctx;
 
   if (mLoadGroup) mLoadGroup->AddRequest(this, nullptr);
 
   return NS_OK;
 }
 
-NS_IMETHODIMP
-nsWyciwygChannel::AsyncOpen2(nsIStreamListener *aListener) {
-  nsCOMPtr<nsIStreamListener> listener = aListener;
-  nsresult rv =
-      nsContentSecurityManager::doContentSecurityCheck(this, listener);
-  if (NS_FAILED(rv)) {
-    mIsPending = false;
-    mCallbacks = nullptr;
-    return rv;
-  }
-  return AsyncOpen(listener, nullptr);
-}
-
 //////////////////////////////////////////////////////////////////////////////
 // nsIWyciwygChannel
 //////////////////////////////////////////////////////////////////////////////
@@ -572,10 +555,9 @@
   nsresult rv;
 
   nsCOMPtr<nsIStreamListener> listener = mListener;
-  nsCOMPtr<nsISupports> listenerContext = mListenerContext;
 
   if (listener) {
-    rv = listener->OnDataAvailable(this, listenerContext, input, offset, count);
+    rv = listener->OnDataAvailable(this, nullptr, input, offset, count);
   } else {
     MOZ_ASSERT(false, "We must have a listener!");
     rv = NS_ERROR_UNEXPECTED;
@@ -599,10 +581,9 @@
        request));
 
   nsCOMPtr<nsIStreamListener> listener = mListener;
-  nsCOMPtr<nsISupports> listenerContext = mListenerContext;
 
   if (listener) {
-    return listener->OnStartRequest(this, listenerContext);
+    return listener->OnStartRequest(this, nullptr);
   }
 
   MOZ_ASSERT(false, "We must have a listener!");
@@ -621,12 +602,10 @@
   mIsPending = false;
 
   nsCOMPtr<nsIStreamListener> listener;
-  nsCOMPtr<nsISupports> listenerContext;
   listener.swap(mListener);
-  listenerContext.swap(mListenerContext);
 
   if (listener) {
-    listener->OnStopRequest(this, listenerContext, mStatus);
+    listener->OnStopRequest(this, nullptr, mStatus);
   } else {
     MOZ_ASSERT(false, "We must have a listener!");
   }
@@ -731,15 +710,13 @@
 
 void nsWyciwygChannel::NotifyListener() {
   nsCOMPtr<nsIStreamListener> listener;
-  nsCOMPtr<nsISupports> listenerContext;
 
   listener.swap(mListener);
-  listenerContext.swap(mListenerContext);
 
   if (listener) {
-    listener->OnStartRequest(this, listenerContext);
+    listener->OnStartRequest(this, nullptr);
     mIsPending = false;
-    listener->OnStopRequest(this, listenerContext, mStatus);
+    listener->OnStopRequest(this, nullptr, mStatus);
   } else {
     MOZ_ASSERT(false, "We must have the listener!");
     mIsPending = false;